### PR TITLE
Emailsubmission useflags

### PR DESCRIPTION
**Pull request by brong**
Hi Ken, popping this one up for your review before it goes live.  We discovered that cancel was storing UNLINKED which breaks replication and made things sad - so I reviewed and came up with this proposal.  Also talked to Neil about whether we wanted EmailSubmission objects to persist past the submission and he said yes.  We'll probably have to make cyr_expire clean them up eventually, but that's future-us' problem.

### CODE REVIEW

File: imap/jmap_mail_submission.c [hunk 1]
```diff
@@ -909,15 +909,11 @@ static void _emailsubmission_update(struct mailbox *submbox,
...
-                        if (!(record->internal_flags & FLAG_INTERNAL_UNLINKED)) {
-                            /* Already sent */
-                            *set_err = json_pack("{s:s}", "type", "cannotUnsend");
-                        }
-                        else if (!strcmp(strval, "canceled")) {
-                            /* Already canceled */
-                            continue;
-                        }
+                    if (record->system_flags & FLAG_ANSWERED) {
+                        if (!strcmp(strval, "final")) continue;
```
Comment (ksmurchison) [inline imap/jmap_mail_submission.c]: The cannotUnsend error is missing for the "final" case

**Comment by ksmurchison** [comment]
OK, so after testing I found one potential problem:
mailbox_update_email_alarms() is used for both futuresend and snooze.  And since a client can sent \flagged and \answered on emails prior to snoozing, the necessary alarms won't be set due to the new flag checks.

I see two ways around this:
1. Use new internal flags for undoStatus (which don't break replication)
2. Split mailbox_update_email_alarms() into separate cases/functions depending on which mbtype/special-use we are dealing with

File: imap/jmap_mail_submission.c (hunk 1 above)
Comment (brong) [reply imap/jmap_mail_submission.c]: hmm yeah good point.

Comment (ksmurchison) [reply imap/jmap_mail_submission.c]: Already fixed after merge.

**Comment by ksmurchison** [comment]
I chose option #2 after merging

**Pull request merged**